### PR TITLE
Improve how the root container is displayed and handled in the blueprint tree

### LINKED ISSUE #4909: Present the root container as the top-level viewport container and differentiate it from other container
The Blueprint Tree is currently presented with the root container at the top of the hierarchy:

image

This causes two issues:
- The root container being, by definition, always alone at its level, it causes significant right-ward indentation for not much user benefit.
- Since nothing can be inserted after or before the root container (again, by definition), it behaves somewhat unexpected with the drag and drop introduced in #4910.

In order to acknowledge the fundamental difference between the root container and all other container, and adjust the presentation to highlight that different and address the current shortcoming, we propose the following changes:
1) The root container no longer has the the collapsing triangle and thus cannot be collapsed. This allows left-shifting the entire tree.
2) The root container is named differently than other containers, for example "Viewport (Grid)"/"Viewport (Horizontal)"/etc.
3) The root container is no longer draggable (though it is now draggable, it's impossible to find a suitable target drop anyways).

Appart from these changes, the root container would remain selectable/editable as  it currently is (maybe with minor adjustment in the selection panel to acknowledge its difference).

Mock-up:

image

### PR DESCRIPTION
**Pull request by abey79**
### What

This PR changes how the root container is displayed and handled:
- It's labeled as "Viewport", since it defines the top-level organisation of the viewport.
- It's not collapsible, so it doesn't have a triangle. This reduces the right-ward drift.
- It (still) cannot be dragged (but now the difference with other containers is more obvious). It now behaves better when dragged over though.
- It cannot be removed.

Other than that, it still behaves mostly as other containers (in particular, it can be selected and edited).

Some changes were needed in the drag-and-drop code, making the "if-statement-of-death" incrementally more complicated, sadly.

* Fixes #4909

image

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
  * Using newly built examples: [app.rerun.io](https://app.rerun.io/pr/4989/index.html)
  * Using examples from latest `main` build: [app.rerun.io](https://app.rerun.io/pr/4989/index.html?manifest_url=https://app.rerun.io/version/main/examples_manifest.json)
  * Using full set of examples from `nightly` build: [app.rerun.io](https://app.rerun.io/pr/4989/index.html?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG

- [PR Build Summary](https://build.rerun.io/pr/4989)
- [Docs preview](https://rerun.io/preview/f0e2e0f5a8b95c99a75fa09038187289ff3a3173/docs) 
- [Examples preview](https://rerun.io/preview/f0e2e0f5a8b95c99a75fa09038187289ff3a3173/examples) 
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

### CODE REVIEW

File: crates/re_ui/src/drag_and_drop.rs [hunk 1]
```diff
@@ -2,21 +2,35 @@
 //!
 //! Works well in combination with [`crate::list_item::ListItem`].
 
+pub enum ItemKind<ItemId: Copy> {
```
Comment (Wumpf) [inline crates/re_ui/src/drag_and_drop.rs:5]: neat, very tidy

File: crates/re_viewport/src/viewport_blueprint_ui.rs [hunk 2]
```diff
@@ -68,6 +66,70 @@ impl Viewport<'_, '_> {
...
+        .drop_target_style(self.state.is_candidate_drop_parent_container(&container_id))
+        .label_style(re_ui::LabelStyle::Unnamed)
+        .with_icon(crate::icon_for_container_kind(
+            &container_blueprint.container_kind,
+        ))
+        .with_buttons(|re_ui, ui| {
+            let response = visibility_button_ui(re_ui, ui, parent_visible, &mut visible);
+            visibility_changed = response.changed();
+            response
+        })
```
Comment (Wumpf) [inline crates/re_viewport/src/viewport_blueprint_ui.rs:106]: what's the usecase of a visibility button on the root container? Just for consistency? I'd suggest dropping it

**Review by Wumpf — APPROVED** [review]
I'm new to this code and I must say these are some really pretty ascii arts explaining me what's going on. Very newcomer friendly!

This design makes way more sense and execution looks solid!

File: crates/re_viewport/src/viewport_blueprint_ui.rs (hunk 2 above)
Comment (abey79) [reply crates/re_viewport/src/viewport_blueprint_ui.rs:106]: That.... is a good point. I'll admit not having thought that through 🙈  Who cares about blanking the viewport?

**Pull request merged**